### ISSUE TITLE
v1-alpha release

### ISSUE THREAD

**Issue by austinletson** [leanprover/lean-action#4]
- [x] Align on features to include in `leanprover/lean-action@v1-alpha
- [x] Create `release/v1-alpha` branch for release candidate
- [x] Test `release/v1-alpha` on existing test projects
- [x] Create release of `release/v1-alpha` with `v1-alpha` tag
- [x] Verify action with `uses: leanprover/lean-action@v1-alpha`

**Comment by austinletson** [comment]
Depends on #1

**Comment by austinletson** [comment]
`lean-action` now supports the features outlined in the initial leanprover/lean4#3950 issue (with the exception of periodic `lake update && lake build` which is out of scope for v1).

I have created a [release/v1-beta](https://github.com/leanprover/lean-action/tree/release/v1-beta) branch to beta test on existing projects.

**Comment by austinletson** [comment]
# Test runs for release/v1-alpha
I ran a series of basic functional tests on `lean-action@release/v1-alpha` across three test projects. I have broken up the tests below by test project and then by test case. There is also a link to the corresponding build where `lean-action` ran and a brief description of the results.

Note, it would be great to automate this type of testing. I bet there is a tool for that.

## [use-lean-standard-action-with-bare-project](https://github.com/austinletson/use-lean-standard-action-with-bare-project)
simple lean package with v4.8.0-rc1
### test case 1: happy build
#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: false
```

#### Results
- [build link](https://github.com/austinletson/use-lean-standard-action-with-bare-project/actions/runs/9051351648/job/24867781429)
- build ran successfully
### test case 2: build fails due to syntax error
Introduce an obvious lean syntax error to test the case where the build will fail
`lean-action` config is same as above.

#### Results
- [build link](https://github.com/austinletson/use-lean-standard-action-with-bare-project/actions/runs/9051538875/job/24868162480)
- build failed due to syntax error
### test case 3: lean4checker
#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: false
    lean4checker: true
```

#### Results
- [build link](https://github.com/austinletson/use-lean-standard-action-with-bare-project/actions/runs/9051390259/job/24867859599)
- build ran successfully
- lean4checker ran successfully


### test case 4: check-reservoir-eligibility
The use-lean-standard-action-with-bare-project should be eligible for reservoir because it is public, has an OSI approved license, and has 2 stars.

#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: false
    check-reservoir-eligibility: true
```

#### Results
- [build link](https://github.com/austinletson/use-lean-standard-action-with-bare-project/actions/runs/9052022036/job/24869229344)
- build ran successfully
- reservoir eligibility was correctly detected
## [use-lean-action-with-toml-lake-file](https://github.com/austinletson/use-lean-action-with-toml-lake-file)
lean package downstream of `mathlib` which uses `lakefile.toml`
### test case 1: happy build
#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: false 
```

#### Results
- [build link](https://github.com/austinletson/use-lean-action-with-toml-lake-file/actions/runs/9044454023/job/24853119978)
- `mathlib` dependency was successfully detected and `lake exe cache get` was run
- build succeeded
### test case 2: lint-module, check-reservoir-eligibility
#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: false 
    lint-module: "UseLeanActionWithTomlLakeFile"
    check-reservoir-eligibility: true
```
#### Results
- [build link](https://github.com/austinletson/use-lean-action-with-toml-lake-file/actions/runs/9051304370/job/24867685190) 
- build succeeded
- lint succeeded
- reservoir check failed because no SPDX license and repo as less than 2 stars

## [use-lean-standard-action-with-lake-test](https://github.com/austinletson/use-lean-standard-action-with-lake-test)
lean package with v4.8.0-rc1 with a `@[test_runner]` entry point in lakefile.lean
### test case 1: fake tests pass
Test case where  `@[test_runner]` entry point returns `0` exit status
#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: true
```

#### Results
- [build link](https://github.com/austinletson/use-lean-standard-action-with-lake-test/actions/runs/9051426898/job/24867935709)
- Build succeeded
- fake tests passed
### test case 2: fake tests fail
`@[test_runner]` entry point returns `1` exit status

#### `lean-action` config
```yaml
- uses: leanprover/lean-action@release/v1-alpha
  with:
    test: true
```
#### Results
- [build link](https://github.com/austinletson/use-lean-standard-action-with-lake-test/actions/runs/9051482613/job/24868050326)
- build succeeded
- fake tests failed

**Comment by austinletson** [comment]
Created [v1-alpha release](https://github.com/leanprover/lean-action/releases/tag/v1-alpha) and [tested the `uses: leanprover/lean-action@v1-alpha`](https://github.com/austinletson/use-lean-standard-action-with-bare-project/blob/main/.github/workflows/ci.yml#L15) syntax.
